src/data/move.ts

```diff
@@ -261,6 +257,18 @@ export abstract class Move implements Localizable {
     return false;
   }
 
+  isAttackMove(): this is AttackMove {
+    return this.category === MoveCategory.PHYSICAL || this.category === MoveCategory.SPECIAL;
+  }
+
+  isStatusMove(): this is StatusMove {
+    return this.category === MoveCategory.STATUS && this.moveTarget !== MoveTarget.USER;
```

Comment:
```suggestion
    return this.category === MoveCategory.STATUS;
```
(Since `SelfStatusMove` extends `StatusMove` now)